### DISCUSSION TITLE
How to use Tailwind with dynamic classes

### DISCUSSION THREAD

**Discussion by tylerism** [tailwindlabs/tailwindcss#8621] [Help]
We want to use Tailwind but Tailwind dynamically generates the css needed by scanning the template files. What about the case where the classes for a component are dynamic? I might have 1 of 3 classes that I want to use in tailwind but the exact class is determined by the properties passed into the component. How is Tailwind supposed to handle this?

**Comment by NovusOpus** [comment]
Check this info in the docs: [Tailwind CSS Docs - Dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names).

**Comment by Chandler-Zhu** [comment]
This is a [demo](https://jsfiddle.net/s4ft6ebg/) I made with [Alpine.js](https://alpinejs.dev/directives/bind) , the `fontSize` property has 3 possible values, and the tailwind class changes according to the current value selected

**Comment by kopahead** [comment]
This:

"The most important implication of how Tailwind extracts class names is that it will only find classes that exist as complete unbroken strings in your source files.

If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:"

Made v3 less dynamic for me. So I decided to stick to 2.2.16

At least I can have my users pick a color they like on the interface and have my theme render (concatenate it to a tailwind class) in real-time.

I use to be able to do this:

```
const someColors = ['red', 'green', 'blue'];

const generateNiceTheme = computed((color) => `bg-${color}-300 border-${color}-600 text-${color}-300`)

<button class="generateNiceTheme(red)" />
```

Can't do that with newer version of tailwind. So stick to old.
